src/pokemon.c

```diff
@@ -5104,7 +5119,16 @@ static void BufferStatRoseMessage(s32 arg0)
 {
     gBattlerTarget = gBattlerInMenuId;
     StringCopy(gBattleTextBuff1, gStatNamesTable[sStatsToRaise[arg0]]);
-    StringCopy(gBattleTextBuff2, gText_StatRose);
+    if (B_X_ITEMS_BUFF == GEN_7)
+    {
+        StringCopy(gBattleTextBuff2, gText_EmptyString3);
+        StringAppend(gBattleTextBuff2, gText_StatSharply);
+        StringAppend(gBattleTextBuff2, gText_StatRose);
```

Comment:
Take a look at `battle_util.c`'s `ItemBattleEffects` for the `HOLD_EFFECT_RANDOM_STAT_UP` case. It shows a nicer looking way of doing it and it doesn't require changing `sText_StatSharply` to `gText_StatSharply` I think.

Reply:
I would rather stick to this StringCopy+StringAppend combo tbh.

The way in which `HOLD_EFFECT_RANDOM_STAT_UP` handles it looks ugly, rather painful and perhaps a bit harder to understand. It just looks weird all around to me :eyes: